client/my-sites/checkout/index.js

```diff
@@ -16,13 +16,31 @@ import {
 	conciergeQuickstartSession,
 } from './controller';
 import SiftScience from 'lib/siftscience';
-import { makeLayout, render as clientRender } from 'controller';
+import { makeLayout, redirectLoggedOut, render as clientRender } from 'controller';
 import { noSite, siteSelection } from 'my-sites/controller';
 import config from 'config';
+import userFactory from 'lib/user';
 
 export default function() {
 	SiftScience.recordUser();
 
+	const user = userFactory();
+	const isLoggedOut = ! user.get();
+
+	if ( isLoggedOut ) {
+		page(
+			'/checkout/:site/add-quickstart-session/:receiptId?',
+			conciergeQuickstartSession,
+			makeLayout,
+			clientRender
+		);
```

Comment:
If the paths with `/add` and `/:receiptId` require login and only the root path supports logged-out view, then this handler is incorrect: navigating directly to one of these two URLs won't redirect to login.

Reply:
Fixed in [9105ae3](https://github.com/Automattic/wp-calypso/pull/33749/commits/9105ae36892a7c9b5d76260f39c9fecc8579f272).